src/metrics/filters/filter.go

```diff
@@ -79,6 +81,21 @@ type Filter interface {
 	Clone() Filter
 }
 
+// TagsFilter matches a string of tags against certain conditions.
+type TagsFilter interface {
```

Comment:
unfortunately this change required some surgery in the `filter` pkg in order to thread the tag iterator all the way down. I think it's fine to break the common `Filter` interface, since we don't actually require it and tags are very special.

Reply:
Agreed, yeah this makes sense.